src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -303,12 +307,23 @@ private void createStudyAsync(UUID caseUuid, String userId, BasicStudyInfos basi
 
     public BasicStudyInfos createStudy(MultipartFile caseFile, String userId, UUID studyUuid) {
         BasicStudyInfos basicStudyInfos = StudyService.toBasicStudyInfos(insertStudyCreationRequest(userId, studyUuid));
-
-        studyServerExecutionService.runAsync(() -> createStudyAsync(caseFile, userId, basicStudyInfos));
+        // Using temp file to store caseFile here because multipartfile are deleted once the request using it is over
+        // Since the next action is asynchronous, the multipartfile could be deleted before being read and cause exceptions
+        File tempFile;
+        try {
+            tempFile = File.createTempFile("tmp_", caseFile.getOriginalFilename());
+            caseFile.transferTo(tempFile);
+        } catch (IOException e) {
+            LOGGER.error(e.toString(), e);
+            throw new StudyException(STUDY_CREATION_FAILED);
```

Comment:
` emitStudyCreationError(studyUuid, userId, e.getMessage()); `?

Reply:
Is it mandatory since the initial API call would already return an error ?